### ISSUE TITLE
Slow image data fetching from Haneke?

### ISSUE THREAD

**Issue by hifall** [Haneke/HanekeSwift#337]
Hi,

I am using Haneke to manage images in table view cells. And I have this observation:

1 create a couple of dozen table cells with an image in each one of them; I call hnk_setImageFromURL with image's url and a placeholder image for each cell;
2 scroll up and down quickly, and I see placeholder images showing for images, even after these images have been loaded. And very shortly after (~sub 0.5s), the placeholder images are replaced with the "real" images.

What we desire is after images have been fetched, they show instantly with scrolling. But the observation indicates there is a delay here.

What is the problem?

**Comment by kwstasna** [comment]
I have the same issu with every Image caching library that is available, even with NSCache.
The problem is that the cells are `reusable` so it means that if you have 30 cells and you see the 15nth, the first 10 cells will be deleted by your memory because of iOS dont want you to use extensivly the memory. So when you scroll to the top it will load again the cells.

Although some apps like "Kitchen Stories" have managed to solve this issue.
I've emailed them to give me some guidance through this problem and they sended me this

> Just make sure to call `collectionView.dequeueReusableSupplementaryViewOfKind` to get the supplementary view. 
> So the collection view does not create a new view every time.

I've tried to do it but with no luck.
Does any expertee here knows what to do to solve this issue?
